src/pages/index.js

```diff
@@ -41,62 +182,136 @@ editProfileMOB.addEventListener("click", () => {
+      console.log(err);
+    })
+    .finally(() => {
+      submitButton.textContent = "Create";
+    });
 }
 
-const handleImageClick = (cardData) => {
-  maxImagePopup.open(cardData);
-};
+//Handles card deletion
+let cardtoDeleteData;
 
-//***************************************************************************************************************************//
-//                                                     Objects                                                               //
-//***************************************************************************************************************************//
+//Handles confirm delete and sets delete data
+function handleConfirmDelete(data) {
+  confirmDeletePopup.open();
+  cardtoDeleteData = data;
+}
 
-const cardSection = new Section(initialCards, createCard, "#cards-list");
-const maxImagePopup = new PopupWithImage("#max-image-modal");
-const editProfilePopup = new PopupWithForm(
-  "#edit-profile-modal",
-  handleEditProfile
-);
-const addPlacePopup = new PopupWithForm("#add-card-modal", handleAddCardSubmit);
+//Deletes card
+async function handleDeleteCard() {
+  try {
+    api.deleteCard(cardtoDeleteData._id);
+    const card = document.querySelector(`li[id='${cardtoDeleteData._id}'`);
+    card.remove();
+  } catch (err) {
+    console.log(err);
+  } finally {
+    confirmDeletePopup.close();
+  }
+}
 
-const userInfo = new UserInfo({
-  nameSelector: ".profile__name",
-  jobSelector: ".profile__subtitle",
-});
+//Changes Profile Picture
+async function handleEditPicSubmit({ link }) {
+  const submitButton = editProfilePicModal.querySelector(
+    "#edit-pic-modal-submit-button"
+  );
+  submitButton.textContent = "Saving...";
 
-//***************************************************************************************************************************//
-//                                             Validation and Rendering                                                      //
-//***************************************************************************************************************************//
+  try {
+    api.changeProfilePicture(link);
+  } catch (err) {
+    console.log(err);
+  } finally {
+    profilePic.src = link;
+    profilePicPopup.close();
+    submitButton.textContent = "Save";
+  }
+}
 
-const editProfileFormValidator = new FormValidator(config, editProfilePopup.popupForm);
+//Handles card Like
+async function handleLikeCard(cardId, method) {
+  const card = document.querySelector(`li[id='${cardId}'`);
+  const cardLike = card.querySelector("#card-like");
+  const cardLikeCounter = card.querySelector("#card-like-counter");
+  try {
+    if (method === "PUT") {
+      api.likeCard(cardId, method);
+      cardLikeCounter.textContent = "1";
+      cardLike.classList.add("card__like_liked");
+    } else {
+      api.removeLike(cardId, method);
+      cardLikeCounter.textContent = "0";
+      cardLike.classList.remove("card__like_liked");
```

Comment:
Now only class `Card` knows every **card selector**, so you need to encapsulate the likes logics inside its methods and call them in `index.js`. It’s a violation of **Single responsibility** principle